src/components/Visualization/Visualization.js

```diff
@@ -215,6 +218,17 @@ export const Visualization = (props) => {
     )
     const { setChart } = useChartContext()
 
+    const visualization = useMemo(() => {
+        const currentClone = cloneDeep(current)
+
+        currentClone &&
+            new Set(Object.keys(options))
```

Comment:
This removes all the options that are not relevant for the visualization type.
Because all the options are at the root level of the object it's hard to filter the options otherwise without affecting other non-option keys.

Reply:
Extracted the logic into a function and added unit tests.